### ISSUE TITLE
native_op.cpp does not compile with TF >= 2.20 (perftools::gputools and Stream::ThenBlas* removed)

### ISSUE THREAD

**Issue by albertz** [rwth-i6/returnn#1833]
`returnn/native_op.cpp` does not compile with TF >= 2.20, so every native op
(`FastBaumWelch`, `GetCtcFsaFastBw`, `NativeLstm2`, ...) fails to build at runtime with
`subprocess.CalledProcessError` from `native_code_compiler`.

Two StreamExecutor APIs it uses have been removed:

1. The `perftools::gputools` namespace is gone (now `stream_executor`). 10 uses, in
   `AsDeviceMemory`, `get_transpose`, and the batched GEMM helper.
2. `Stream::ThenBlasGemm` / `Stream::ThenBlasGemmBatched` are gone. The replacement is to go
   through `BlasSupport` directly: `stream->parent()->AsBlas()->DoBlasGemm(...)`, which also
   changes the return convention (`absl::Status` / `bool` instead of a chainable status) and
   takes `alpha`/`beta` by pointer with an explicit element type.

Neither symbol appears anywhere in the TF 2.20 headers.

### Reproduce

```python
from returnn.tf.native_op import ctc_loss

ctc_loss(logits=logits, logits_seq_lens=..., logits_time_major=True,
         targets=..., targets_seq_lens=..., blank_index=V - 1)
```

First errors from nvcc (~107 in total, the rest cascade from the first):

```
native_op.cpp(263): error: name followed by "::" must be a class or namespace name
native_op.cpp(263): error: this declaration has no storage class or type specifier
native_op.cpp(378): error: identifier "context" is undefined
native_op.cpp(379): error: "perftools" has already been declared in the current scope
native_op.cpp(384): error: identifier "get_transpose" is undefined
```

### Impact

Any setup relying on a native op on TF >= 2.20. It also matters for the RF TF backend: without
the native op, `TFBackend.ctc_loss` has to fall back to `tf.nn.ctc_loss`, and neither TF
implementation is usable at production scale for large vocabularies --

- the classic `CTCLoss` op is **CPU-only** (`kernels.get_registered_kernels_for_op("CTCLoss")`
  reports `['CPU']`), so both CTC heads leave the GPU every step;
- the dense path (`ctc_loss_dense`) broadcasts to `[T, B, S, V]` and crashes with
  `CUDA_ERROR_ILLEGAL_ADDRESS` once that exceeds 2^31 elements, which for vocab 10241 happens at
  batch size 40 (reported upstream: tensorflow/tensorflow#124949).

Measured on a production conformer AED+CTC training: the CPU-only CTC accounts for most of a
~2.1-2.3x wall-clock gap against the equivalent PyTorch-backend run.

### Environment

TF 2.20.0 (pip), CUDA 12.5.1 build / 12.6.3 nvcc, cuDNN 9, Python 3.12.3, NVIDIA H100,
driver 570.211.01, RHEL 9.

### Fix

I have a patch and will open a PR. It guards the new code behind
`TF_MAJOR_VERSION`/`TF_MINOR_VERSION` at the version actually verified (>= 2.20), so older TF
keeps the exact code it has today and cannot regress; a namespace alias (`returnn_se`) covers
point 1 and resolves identically on older versions.

With the patch on TF 2.20:

- the op compiles cleanly (0 errors);
- `native_op.ctc_loss` agrees with `tf.compat.v1.nn.ctc_loss`: max |loss diff| 0.0,
  max |grad diff| 8.9e-07;
- `tests/test_TFNativeOp.py` CPU-only: **52 passed, 7 skipped**;
- `tests/test_TFNativeOp.py` GPU: **all pass with `NVIDIA_TF32_OVERRIDE=0`**.

### TF32

The new `DoBlasGemm` takes `NumericOptions`, whose default is `allow_tf32=true`. TF32 costs about
5e-5 to 1.4e-4 relative on a float32 GEMM, which is enough to fail `test_native_lstm2_grad` and
`test_native_lstm2_grad_start_step` on an Ampere+ GPU, on either side of the comparison.

Resolved in two parts:

- the op passes `tensorflow::tensor_float_32_execution_enabled()`, so it follows TF's global TF32
  setting exactly as TF's own kernels do, rather than hardcoding a choice;
- `tests/test_TFNativeOp.py` calls `tf.config.experimental.enable_tensor_float_32_execution(False)`
  before creating its session. These tests compare native ops against TF references for
  exactness, so both sides should compute in plain float32.

With that, on default env and no `NVIDIA_TF32_OVERRIDE`:

```
test_TFNativeOp.py  GPU:       52 passed, 7 skipped
test_TFNativeOp.py  CPU-only:  52 passed, 7 skipped
```
